spring-xd-runner/src/main/java/org/springframework/bus/xd/bootstrap/ModuleOptionsPropertySourceInitializer.java

```diff
@@ -71,7 +71,7 @@ public void initialize(ConfigurableApplicationContext applicationContext) {
 		ModuleOptionsMetadata resolved = resolver.resolve(getModuleDefinition());
 		Map<String, Object> map = new LinkedHashMap<String, Object>();
 		for (ModuleOption option : resolved) {
-			map.put(option.getName(), option.getDefaultValue());
+			map.put(option.getName(), (option.getDefaultValue() == null) ? "" : option.getDefaultValue());
```

Comment:
Instead of putting empty string values in the map, would it work if we do not include an option in the map at all if it does not have a non-null default value and no value has been explicitly provided?